ft1d: fix room number length check

goldstar611 opened the pull request:
The ft1d driver would throw an error when editing the Wires-X room number if the room number was less than 5 digits because there was no zero padding.

https://chirp.danplanet.com/issues/10831

# CHIRP PR Checklist

The following must be true before PRs can be merged:

* All tests must be passing.
* Commits should be squashed into logical units.
* Commits should be rebased (or simply rebase-able in the web UI) on current master. Do not put merge commits in a PR.
* Commits in a single PR should be related.
* Major new features or bug fixes should reference a [CHIRP issue](https://chirp.danplanet.com/projects/chirp/issues).
* New drivers should be accompanied by a test image in `tests/images` (except for thin aliases where the driver is sufficiently tested already).
* All files must be GPLv3 licensed or contain no license verbiage. No additional restrictions can be placed on the usage (i.e. such as noncommercial).

Please also follow these guidelines:

* Keep cleanups in separate commits from functional changes.
* Please write a reasonable commit message, especially if making some change that isn't totally obvious (such as adding a new model, adding a feature, etc).
* Do not add new py2-compatibility code (No new uses of `six`, `future`, etc).
* All new drivers should set `NEEDS_COMPAT_SERIAL=False` and use `MemoryMapBytes`.
* New drivers and radio models will affect the Python3 test matrix. You should regenerate this file with `tox -emakesupported` and include it in your commit.

kk7ds commented on chirp/drivers/ft1d.py:
```diff
@@ -1966,9 +1966,9 @@ def apply_WiresX_category(cls, setting, obj):
         setattr(obj, "name", val)
 
-    def apply_WiresX_roomid(cls, setting, obj):
+    def apply_WiresX_roomid(self, setting, obj):
```
Yeah, this (and others) should be `self` not `cls`.. Weird.

goldstar611 replied:
I started down the path of fixing other instances but I realized some of the functions were modifying class-scoped variables, so in that aspect, it made sense logically. Plus it had nothing to do with the issue at hand so I only changed this one.

kk7ds replied:
Ack, there's almost no good reason for keeping per-class state so those are almost definitely bugs waiting to be found, but fair enough.

kk7ds commented:
IDK about all the things you're zero-padding, but otherwise looks okay to me. I'll hold off on this for a bit as there are a couple other things pending and I'll try to merge all together to avoid a single-small-fix build. Thanks!

goldstar611 commented:
> IDK about all the things you're zero-padding, but otherwise looks okay to me. I'll hold off on this for a bit as there are a couple other things pending and I'll try to merge all together to avoid a single-small-fix build. Thanks!

I started on a one line fix for the Wires-X zero padding issue but figured there must be an existing zero_pad function to support the other fields. It was `backtrack_zero_pad` so I renamed it to simply `zero_pad` and updated the logic while I was there. This is just function renaming fallout.

Thanks for reviewing and the feedback!

kk7ds commented:
Oh right, I see, duh. I was just focused on the room change, but glad it's even less change than I was thinking.

goldstar611 commented:
> Oh right, I see, duh. I was just focused on the room change, but glad it's even less change than I was thinking.

It's crazy how a one line change can end up changing 10 more.
